x/distribution/keeper/delegation.go

```diff
@@ -99,14 +99,18 @@ func (k Keeper) calculateDelegationRewards(ctx sdk.Context, val sdk.Validator, d
 	currentStake := val.TokensFromShares(del.GetShares())
 	if stake.GT(currentStake) {
 		// account for rounding errors due to stake being multiplied by slash fractions
-		currentStakeRoundUp := val.TokensFromSharesRoundUp(del.GetShares())
+		currentStakeRoundUp := sdk.MaxDec(
```

Comment:
Do you know how or why this happens @rigelrozanski? Would it possible to maybe add a comment explaining this?

Reply:
good idea - I've beefed up the comment here, after further reflection, we actually only ever need to consider a single decimal place greater, and should not be concerned about using MaxDec, as it should always lead to the same result